libraries/AP_ESC_Telem/AP_ESC_Telem.cpp

```diff
@@ -607,6 +657,29 @@ void AP_ESC_Telem::update()
                     error_rate  : rpmdata.error_rate
                 };
                 AP::logger().WriteBlock(&pkt, sizeof(pkt));
+
+#if AP_EXTENDED_ESC_TELEM_ENABLE
+                // Write ESC extended status messages
+                //   id: starts from 0
+                //   input duty: duty cycle input to the ESC in percent
+                //   output duty: duty cycle output to the motor in percent
+                //   status flags: manufacurer-specific status flags
+                bool has_ext_data = telemdata.types & 
+                        (AP_ESC_Telem_Backend::TelemetryType::INPUT_DUTY |
+                         AP_ESC_Telem_Backend::TelemetryType::OUTPUT_DUTY |
+                         AP_ESC_Telem_Backend::TelemetryType::FLAGS);
+                if (has_ext_data) {
+                    const struct log_EscX pkt2{
+                        LOG_PACKET_HEADER_INIT(uint8_t(LOG_ESCX_MSG)),
+                        time_us     : AP_HAL::micros64(),
+                        instance    : i,
+                        input_duty  : telemdata.input_duty,
+                        output_duty : telemdata.output_duty,
+                        flags       : telemdata.flags,
+                    };
+                    AP::logger().WriteBlock(&pkt2, sizeof(pkt2));
```

Comment:
use WriteStreaming instead, for less overhead in the code

Reply:
Implemented WriteStreaming instead of WriteBlock